src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,73 +6,145 @@
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.datamodel.models.input.connector.`type`.{
   LineTypeInput,
   Transformer2WTypeInput
 }
 import edu.ie3.osmogrid.ActorStopSupport
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
-import edu.ie3.osmogrid.guardian.OsmoGridGuardian
-import edu.ie3.osmogrid.io.input.InputDataProvider.Request
-import edu.ie3.osmogrid.lv.coordinator.LvCoordinator.stopBehavior
-import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
-import org.slf4j.Logger
+import edu.ie3.osmogrid.io.input.InputDataProvider.InputDataEvent
+import edu.ie3.osmogrid.model.{OsmoGridModel, SourceFilter}
+import edu.ie3.util.osm.model.OsmEntity.{Node, Relation, Way}
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
 
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try, Using}
 import java.util.UUID
 
-object InputDataProvider extends ActorStopSupport[Request] {
+object InputDataProvider extends ActorStopSupport[InputDataEvent] {
 
+  // external requests
   sealed trait Request
 
   final case class ReqOsm(
-      runId: UUID,
       replyTo: ActorRef[InputDataProvider.Response],
-      filter: PbfFilter
+      filter: SourceFilter
   ) extends Request
+      with InputDataEvent
 
   final case class ReqAssetTypes(
       runId: UUID,
       replyTo: ActorRef[InputDataProvider.Response]
   ) extends Request
+      with InputDataEvent
 
-  object Terminate extends Request
+  case object Terminate extends Request with InputDataEvent
 
   // external responses
   sealed trait Response
-  final case class RepOsm(runId: UUID, osmModel: OsmoGridModel) extends Response
-  final case class InvalidOsmRequest(reqRunId: UUID, actualRunId: UUID)
+  final case class RepOsm(runId: UUID, osmModel: OsmoGridModel)
+      extends Response
+      with InputDataEvent
+  final case class OsmReadFailed(reason: Throwable)
+      extends Response
+      with InputDataEvent
+  final case class RepAssetTypes(assetInformation: AssetInformation)
       extends Response
-  final case class OsmReadFailed(reason: Throwable) extends Response
-  final case class RepAssetTypes(
-      assetInformation: AssetInformation
-  ) extends Response
 
   final case class AssetInformation(
       lineTypes: Seq[LineTypeInput],
       transformerTypes: Seq[Transformer2WTypeInput]
   )
 
-  def apply(cfg: OsmoGridConfig.Input): Behavior[Request] =
-    Behaviors
-      .receive[Request] {
-        case (ctx, ReqOsm(runId, replyTo, filter)) =>
-          ctx.log.warn("Reading of data not yet implemented.")
-          Behaviors.same
-        case (ctx, ReqAssetTypes(_, _)) =>
+  // internal api
+  sealed trait InputDataEvent
```

Comment:
I'd once again raise the question, if the `InputDataEvent` trait is needed. In this implementation, the actor would be able to understand it's own `Response`s (meant to be sent out). This should not be too harmful, but it is less tight, than it could be.

Reply:
`InputDataProvider` is supposed to receive and forward `Response`s, right?

https://github.com/ie3-institute/OSMoGrid/blob/52170672d6a7fc7e7f29ad95c53492cce9e5da28/src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala#L131-L145

Reply:
Oh, yes. Now, I got it!

Somewhat, I don't like this fully. Because to the outside world, the protocol is more open, than it needs to be. However, lets' keep it this way.